In file included from /public/home/zhangxin/external-libraries/numpy/core/include/numpy/ndarraytypes.h:1929,
                 from /public/home/zhangxin/external-libraries/numpy/core/include/numpy/ndarrayobject.h:12,
                 from /public/home/zhangxin/external-libraries/numpy/core/include/numpy/arrayobject.h:5,
                 from zuda_cpu.cpp:1120:
/public/home/zhangxin/external-libraries/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
      |  ^~~~~~~
Traceback (most recent call last):
  File "/public/home/zhangxin/zuda/test/./test_zuda_cpu.py", line 13, in <module>
    init()
  File "/public/home/zhangxin/external-libraries/pyquda/mpi.py", line 55, in init
    cuda.Device(gpuid).use()
  File "cupy/cuda/device.pyx", line 192, in cupy.cuda.device.Device.use
  File "cupy/cuda/device.pyx", line 198, in cupy.cuda.device.Device.use
  File "cupy_backends/cuda/api/runtime.pyx", line 374, in cupy_backends.cuda.api.runtime.setDevice
  File "cupy_backends/cuda/api/runtime.pyx", line 143, in cupy_backends.cuda.api.runtime.check_status
cupy_backends.cuda.api.runtime.CUDARuntimeError: cudaErrorNoDevice: no CUDA-capable device is detected
